inst/include/RcppArmadilloAs.h

```diff
@@ -32,10 +32,17 @@ namespace traits {
         Exporter(SEXP x) : data(x){}
 
         inline arma::field<T> get() {
-            size_t n = data.size() ;
-            arma::field<T> out( n ) ;
-            for(size_t i=0; i<n; i++){
-                out[i] = as<T>(data[i]) ;
+            arma::ivec dims = data.attr("dim");
+            dims.resize(3);
```

Comment:
Just thought, based on re-reading http://arma.sourceforge.net/docs.html#field : we don't actually know that `dims` has length three.   We could have a `field` of `vec`, or `mat`, or `cube`.

Reply:
The dimensions of the field are separate from the dimensions of the objects the field contains.  Each object with the field (eg. vector, matrix, cube, ...) can have its own unique size.

Say we have a 3D field of matrices: `field<mat> F(3,4,5)`. Within F, the matrix at F(0,0,0) could have a size of (2,3), while the matrix at F(1,2,3) could have a a size of (100,200).

Reply:
Of course, like @eddelbuettel said, the field itself could be 1D, 2D or 3D.